### DISCUSSION TITLE
Need help with REST API connector workflow

### DISCUSSION THREAD

**Discussion by BloemenBezorgen** [Budibase/budibase#7781] [Help]
Hi there, I have a beginners question.

I would like to create an app to edit products in my Woocommerce shop using REST API calls. 

These are the first screens that I have created so far:
/products
/products/:id

the /products screen is a list with all the products. I use this call to populate the screen: REST API call: GET https://woocommerce-470197-2486223.cloudwaysapps.com/wp-json/wc/v3/products/ 

This is the screen:
![Product list](https://user-images.githubusercontent.com/82037117/190382700-4f4fc845-cba0-442a-a761-57b465fc7fee.png)

In each row there is a link to the products/:id screen 

The /products/:id should be populated with the data from one product. Should I setup a new API call to call the data from that single product? Or will budibase use the data fro the https://woocommerce-470197-2486223.cloudwaysapps.com/wp-json/wc/v3/products/  call?

**Comment by melohagan** [comment]
Hey @BloemenBezorgen 

When you link to a new screen, the data will be retrieved independently on that screen, so you probably should have a query that can get the detail of a single product. 

If there's any benefit to it, technically you could pass row data in the URL and parse it on the other side.
https://docs.budibase.com/docs/url-parameters

But it probably won't be as useable as using a data provider in the new 'detail' screen.

**Reply by BloemenBezorgen** [reply]
Thank you Melohagan. I will then set up an API call for the single product: ending with product/{{id}} to set up the form first so I know it works. After that I will try to add the row data to the url from the products list.

**Reply by BloemenBezorgen** [reply]
I have one more question. From the product edit screen, after editing the product data,  should I set up PUT rest api call in automations? So when. A product is updated > a rest api put call is send to woocommerce with the adjusted data?

**Reply by melohagan** [reply]
You don't necessarily need to use an automation. You could use the **Execute Query** action: https://docs.budibase.com/docs/referencing

**Reply by BloemenBezorgen** [reply]
Thank you! I will dive into that documentation.
